pkg/kubectl/resource_printer.go

```diff
@@ -394,9 +394,21 @@ func printService(svc *api.Service, w io.Writer) error {
 		formatLabels(svc.Spec.Selector), svc.Spec.PortalIP, svc.Spec.Ports[0].Port, svc.Spec.Ports[0].Protocol); err != nil {
 		return err
 	}
-	for i := 1; i < len(svc.Spec.Ports); i++ {
+	count := len(svc.Spec.Ports)
+	if len(svc.Spec.PublicIPs)+1 > count {
```

Comment:
Uncomfortably easy to introduce an off-by-one error here. I followed through the logic and it looks correct, but you should add a unit test.